core/unix/os.c

```diff
@@ -1301,6 +1301,28 @@ os_slow_exit(void)
     IF_NO_MEMQUERY(memcache_exit());
 }
 
+/* Helper function that calls cleanup_and_terminate after blocking most signals
+ *(i#2921).
+ */
+void
+cleanup_and_terminate_helper(dcontext_t *dcontext, int sysnum, ptr_uint_t sys_arg1,
```

Comment:
Please pick a more descriptive name: something like "block_cleanup_and_terminate" or "cleanup_and_terminate_blocked" or sthg which is more readable than "helper" which does not add anything to aid in understanding the code.  This is exported from this file which makes the name even more important.

Reply:
Done.